libvast/src/system/index.cpp

```diff
@@ -655,14 +668,21 @@ void index_state::add_partition_creation_listener(
 // -- introspection ----------------------------------------------------------
 
 void index_state::send_report() {
-  VAST_ASSERT(workers == idle_workers.size() + busy_workers.size());
-  auto msg = report{.data = {
-                      {"query.backlog.normal", backlog.normal.size()},
-                      {"query.backlog.low", backlog.low.size()},
-                      {"query.workers.idle", idle_workers.size()},
-                      {"query.workers.busy", busy_workers.size()},
-                    }};
-  self->send(accountant, msg);
+  auto counters = std::exchange(this->counters, {});
+  auto msg = report{
+    .data = {
+      {"query.pending.partitions", pending_queries.num_partitions()},
+      {"query.pending.queries", pending_queries.num_queries()},
+      {"query.partition.materializations", counters.partition_materializations},
+      {"query.partition.lookups", counters.partition_lookups},
```

Comment:
This is independent of the query, so it should probably be under the `index` category and without the metadata.

Reply:
I'm not sure I understand, this report doesn't have any metadata attached? And we use the `query.` prefix for closely related metrics like the worker counts. I really don't like to add them to an `index` category, because I think the evaluation of the non-active parts of the database should be moved to a different component eventually anyways.

Reply:
I assumed that the new `query.*` metrics carry the same query ID as all the pre-existing metrics in the same category. Everything else would be unintuitive to the user. Maybe we can use the `scheduler.*` prefix here to anticipate the index -> scheduler + partition-builder split?

Reply:
Should we rename the others in that block as well then?

Reply:
I think that makes sense as they're also not tied to a particular query.